share/functions/diff.fish

```diff
@@ -0,0 +1,6 @@
+# Use colours in diff output, if supported
+if command diff --color=auto /dev/null /dev/null 2>&1
```

Comment:
```suggestion
if command -vq diff; and command diff --color=auto /dev/null{,} >/dev/null 2>&1
```

Some more paranoia, sorry :smile_cat: - we explicitly mention `diff` as requirement in CONTRIBUTING so it's probably not installed on every system.

(I can merge and squash that, no need to push again).

Reply:
Sure, fine by me. :+1: